
### Nama : Muchammad Yuda Tri Ananda 
### NIM  : 24060124110142
### Kelas  : D



Untuk mempelajari lebih lanjut mengenai properti dari nilai mean dan median, silahkan terlebih 
dahulu download dataset (“Latihan.csv”) dari link berikut ini: 

- https://s.id/aDGce  

Dataset “Latihan” merupakan dataset yang berisi sampel jumlah jawaban benar untuk masing
masing tipe soal dan skornya pada suatu Latihan untuk suatu mata kuliah tertentu.



Aturan Penulisan Kode Program: 
- Gunakan Google Collaboratory dan library Pandas untuk mengerjakan lembar kerja 
berikut. 
- Simpan DataFrame hasil pembacaan file dataset “Latihan.csv” ke dalam variable “df”



In [74]:
# Import Library
import pandas as pd

# Download File "Latihan.csv"
LINK_DATASET = "https://github.com/myudak/belajarUndip/blob/367836e9ccf8f699c44b920cebfd1b662624693e/SEMESTER_2/Statistika/Latihan.csv?raw=true"

# Simpan DataFrame ke variable df
df = pd.read_csv(LINK_DATASET)

# Isi dataframe top 5 baris
df.head()


Tipe1  Tipe2  Tipe3  Tipe4
0      5      5      4      2
1      5      3      2      3
2      5      4      2      3
3      5      3      2      2
4      4      3      2      4

## <u>Properti #1:</u> Menambah atau mengurangi dengan suatu nilai konstanta untuk semua dataset. 

Untuk menemukan properti pertama, kita akan mencari nilai mean (rata-rata) dan median (nilai 
tengah) dari kumpulan data, lalu menambahkan (atau mengurangi) suatu nilai konstanta yang 
sama untuk setiap baris dalam data, dan menemukan nilai rata-rata dan median yang baru. 

In [75]:
KOLOM = 'Tipe4'
tipe4 = df[KOLOM]
tipe4.head() # Melihatkan top 5 dataframe kolom Tipe4

0    2
1    3
2    3
3    2
4    4
Name: Tipe4, dtype: int64

### Langkah 1: Hitung nilai mean dan median sebelum memodifikasi dataset untuk  variabel **Tipe4** (bebas memilih)

In [76]:
tipe4_mean = tipe4.mean()
tipe4_median = tipe4.median()

print(f"Dataset kolom {KOLOM} sebelum penambahan konstanta")
print('Mean\t:', tipe4_mean)
print('Median\t:', tipe4_median)

Dataset kolom Tipe4 sebelum penambahan konstanta
Mean	: 3.6785714285714284
Median	: 3.5


### Langkah 2: Modifikasi satu buah variabel dari dataset terkait dengan menambahkan atau mengurangi suatu konstanta untuk setiap data. 

In [77]:
KONSTANTA = 69

In [78]:
tipe4_modifikasi = df[KOLOM].map(lambda x: x + KONSTANTA)
tipe4_modifikasi.head()

0    71
1    72
2    72
3    71
4    73
Name: Tipe4, dtype: int64

### Langkah 3: Hitung nilai mean dan median dari modified dataset untuk melengkapi tabel pada Langkah 1 dan lakukan analisa terhadap hasil yang diperoleh sesuai pertanyaan berikut ini. 

In [79]:
tipe4_modifikasi_mean = tipe4_modifikasi.mean()
tipe4_modifikasi_median = tipe4_modifikasi.median()

print(f"Dataset kolom {KOLOM} setelah penambahan konstanta")
print('Mean\t:', tipe4_modifikasi_mean)
print('Median\t:', tipe4_modifikasi_median)

Dataset kolom Tipe4 setelah penambahan konstanta
Mean	: 72.67857142857143
Median	: 72.5


| **Statistical Descriptive** | **Python Code**            | **Original Data**         | **Modified Data**         |
|----------------------------|----------------------------|---------------------------|---------------------------|
| μ (Mean)                   | `tipe4.mean()`             | 3.6785714285714284        | 72.67857142857143         |
| Median                     | `tipe4.median()`           | 3.5                       | 72.5                      |


| Python Code: |       ```df['Tipe4'].map(lambda x: x + 69)```        |
|--------------|---------------|
| Deskripsi:   | Menambahkan sebuah konstanta sebesar **69** untuk setiap data dari variabel **Tipe4** |

| Bagaimana penjumlahan/pengurangan suatu konstanta pada semua data mengubah nilai rata rata dan median?      |
|-----------------------------|
| Menambahkan atau mengurangi suatu konstanta pada semua data akan menggeser seluruh nilai data tanpa mengubah bentuk distribusinya. Akibatnya, nilai rata-rata (mean) dan median akan berubah sebesar konstanta tersebut: jika ditambah $c$, maka mean dan median juga akan bertambah $c$; jika dikurang $c$, maka keduanya akan berkurang sebesar $c$. |

## <u>Properti #2:</u> Mengalikan atau membagi dengan suatu nilai konstanta untuk semua dataset. 
Mirip dengan properti pertama, apa yang akan terjadi ketika kita mengalikan atau membagi 
semua data dengan suatu nilai konstanta.

In [80]:
KOLOM = 'Tipe3'
tipe3 = df[KOLOM]
tipe3.head() # Melihatkan top 5 dataframe

0    4
1    2
2    2
3    2
4    2
Name: Tipe3, dtype: int64

### Langkah 1: Hitung nilai mean dan median sebelum memodifikasi dataset untuk variabel **Tipe3** (bebas memilih)

In [81]:
tipe3_mean = tipe3.mean()
tipe3_median = tipe3.median()

print(f"Dataset kolom {KOLOM} sebelum penambahan konstanta")
print('Mean\t:', tipe3_mean)
print('Median\t:', tipe3_median)

Dataset kolom Tipe3 sebelum penambahan konstanta
Mean	: 5.339285714285714
Median	: 4.0


### Langkah 2: Modifikasi satu buah variabel dari dataset terkait dengan mengalikan suatu konstanta (69) untuk setiap data.  

In [82]:
KONSTANTA = 69

In [83]:
tipe3_modifikasi = df[KOLOM].map(lambda x: x * KONSTANTA) # untuk setiap data dalam kolom 'Tipe3' dikalikan KONSTANTA 69
tipe3_modifikasi.head()

0    276
1    138
2    138
3    138
4    138
Name: Tipe3, dtype: int64

### Langkah 3: Hitung nilai mean dan median dari modified dataset untuk melengkapi tabel pada Langkah 1 dan lakukan analisa terhadap hasil yang diperoleh sesuai pertanyaan berikut ini. 

In [84]:
tipe3_modifikasi_mean = tipe3_modifikasi.mean()
Tipe3_modifikasi_median = tipe3_modifikasi.median()

print(f"Dataset kolom {KOLOM} setelah penambahan konstanta")
print('Mean\t:', tipe3_modifikasi_mean)
print('Median\t:', Tipe3_modifikasi_median)

Dataset kolom Tipe3 setelah penambahan konstanta
Mean	: 368.4107142857143
Median	: 276.0


| **Statistical Descriptive** | **Python Code**            | **Original Data**         | **Modified Data**         |
|----------------------------|----------------------------|---------------------------|---------------------------|
| μ (Mean)                   | `tipe3.mean()`             | 5.339285714285714        | 368.4107142857143         |
| Median                     | `tipe3.median()`           | 4.0                       | 276.0                      |


| Python Code: |       ```df['Tipe3'].map(lambda x: x * 69)```        |
|--------------|---------------|
| Deskripsi:   | Mengalikan sebuah konstanta sebesar **69** untuk setiap data dari variabel **Tipe3** |

| Bagaimana perkalian/pembagian suatu konstanta pada semua data mengubah nilai rata-rata dan median?       |
|-----------------------------|
| Jika semua data dikalikan atau dibagi dengan suatu konstanta $c$, maka **rata-rata (mean)** dan **median** juga akan dikalikan atau dibagi dengan konstanta tersebut. Artinya, jika setiap nilai dalam data dikalikan $c$, maka nilai mean dan median juga berubah menjadi $\text{mean} \times c$ dan $\text{median} \times c$. Begitu pula jika dibagi $c$, keduanya menjadi $\text{mean} \div c$ dan $\text{median} \div c$. Namun, seperti pada penjumlahan/pengurangan, bentuk distribusi tetap sama; hanya skala datanya saja yang berubah. Contoh dalam hal ini Median original adalah 4.0 dan median yang dimodifikasi adalah 276.0 yaitu 69 konstanta kalinya median original 4.0.  |

## <u>Properti #3:</u> Sensitif terhadap Outliers 
Gunakan data yang sama, apa yang akan terjadi jika kita terdapat siswa baru super genius dimana 
untuk setiap tipe ujian selalu mendapatkan poin maksimal yakni 1000. 

In [85]:
KOLOM = 'Tipe3'
tipe3 = df[KOLOM]
tipe3.head()

0    4
1    2
2    2
3    2
4    2
Name: Tipe3, dtype: int64

### Langkah 1: Hitung nilai mean dan median sebelum memodifikasi dataset untuk variabel Tipe3 (bebas memilih)

In [86]:
tipe3_mean = tipe3.mean()
tipe3_median = tipe3.median()

print(f"Dataset kolom {KOLOM} sebelum penambahan konstanta")
print('Mean\t:', tipe3_mean)
print('Median\t:', tipe3_median)

Dataset kolom Tipe3 sebelum penambahan konstanta
Mean	: 5.339285714285714
Median	: 4.0


### Langkah 2: Modifikasi dataset dengan menambahkan satu baris data. 

In [87]:
DATA_BARU = 1000
tipe3_modifikasi = pd.concat([df[KOLOM], pd.Series([DATA_BARU])])
tipe3_modifikasi.tail() # Melihatkan bahwa data baru sudah ditambahkan

52       8
53      12
54       5
55       3
0     1000
dtype: int64

### Langkah 3: Hitung nilai mean dan median dari modified dataset untuk melengkapi tabel pada Langkah 1 dan lakukan analisa terhadap hasil yang diperoleh sesuai pertanyaan berikut ini. 

In [88]:
tipe3_modifikasi_mean = tipe3_modifikasi.mean()
tipe3_modifikasi_median = tipe3_modifikasi.median()

print(f"Dataset kolom {KOLOM} setelah penambahan konstanta")
print('Mean\t:', tipe3_modifikasi_mean)
print('Median\t:', tipe3_modifikasi_median)

Dataset kolom Tipe3 setelah penambahan konstanta
Mean	: 22.789473684210527
Median	: 4.0


| **Statistical Descriptive** | **Python Code**            | **Original Data**         | **Modified Data**         |
|----------------------------|----------------------------|---------------------------|---------------------------|
| μ (Mean)                   | `tipe3.mean()`             | 5.339285714285714        | 127.03508771929825         |
| Median                     | `tipe3.median()`           | 4.0                       | 4.0                      |


| Python Code: |       ```pd.concat([df["Tipe3"], pd.Series([1000])])```        |
|--------------|---------------|
| Deskripsi:   | Menambahkan satu baris data dengan nilai 1000 untuk variabel Tipe3 |

| Bagaimana perkalian/pembagian suatu konstanta pada semua data mengubah nilai rata-rata dan median?       |
|-----------------------------|
| Penambahan satu baris data baru yang berupa nilai ekstrem (misalnya, sangat besar atau sangat kecil dibandingkan dengan nilai lainnya) akan mempengaruhi rata-rata (mean) secara signifikan, tetapi pengaruhnya terhadap median cenderung lebih kecil, tergantung pada posisi nilai ekstrem dalam data. Contoh dalam hal ini mean rata-rata berubah drastis, sedangkan median tidak berubah sama sekali |